### PR TITLE
update logic ExportAppStateAndValidators function

### PR DESCRIPTION
**Pull request by ThanhNhann**
👮🏻👮🏻👮🏻 !!!! REFERENCE THE PROBLEM YOUR ARE SOLVING IN THE PR TITLE AND DESCRIBE YOUR SOLUTION HERE !!!! DO NOT FORGET !!!! 👮🏻👮🏻👮🏻

Close: #920 
# What is the purpose of the change
- Update logic return of `ExportAppStateAndValidators` function
- This is a small thing, but important for the logic code
# PR Checklist:

- [x] Have you read the [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md)?
- [x] Does your PR follow the [C4 patch requirements](https://rfc.zeromq.org/spec:42/C4/#23-patch-requirements)?
- [x] Have you rebased your work on top of the latest master? 
- [x] Have you checked your code compiles? (`make`)
- [x] Have you included tests for any non-trivial functionality?
- [x] Have you checked your code passes the unit tests? (`make test`)
- [x] Have you checked your code formatting is correct? (`go fmt`)
- [x] Have you checked your basic code style is fine? (`golangci-lint run`)
- [ ] If you added any dependencies, have you checked they do not contain any known vulnerabilities? (`go list -json -m all | nancy sleuth`)
- [ ] If your changes affect the client infrastructure, have you run the integration test?
- [ ] If your changes affect public APIs, does your PR follow the [C4 evolution of public contracts](https://rfc.zeromq.org/spec:42/C4/#26-evolution-of-public-contracts)?
- [ ] If your code changes public APIs, have you incremented the crate version numbers and documented your changes in the [CHANGELOG.md](https://github.com/crypto-org-chain/chain-main/blob/master/CHANGELOG.md)?
- [ ] If you are contributing for the first time, please read the agreement in [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md#developer-certificate-of-originn).

Thank you for your code, it's appreciated! :)

### CODE REVIEW

**Review by tomtau — COMMENTED** [review]
@ThanhNhann would you mind opening an issue to detail the problem this is solving?

**Comment by ThanhNhann** [comment]
hi @tomtau, thank you for your reply, I just created the issue for this pr: https://github.com/crypto-org-chain/chain-main/issues/920
plz check

File: app/export.go [hunk 1]
```diff
@@ -40,12 +40,15 @@ func (app *ChainApp) ExportAppStateAndValidators(
 	}
 
 	validators, err := staking.WriteValidators(ctx, app.StakingKeeper)
```
Comment (mmsqe) [inline app/export.go:42]: [err](https://github.com/cosmos/cosmos-sdk/blob/v0.46.7/x/staking/genesis.go#L15) here is actually nil

Comment (ThanhNhann) [reply app/export.go:42]: thank @yihuang for your update

File: app/export.go [hunk 2]
```diff
@@ -40,12 +40,15 @@ func (app *ChainApp) ExportAppStateAndValidators(
...
+	if err != nil { 
+		return servertypes.ExportedApp{}, err
+	}
 	return servertypes.ExportedApp{
 		AppState:        appState,
 		Validators:      validators,
 		Height:          height,
 		ConsensusParams: app.BaseApp.GetConsensusParams(ctx),
-	}, err
+	}, nil
```
Comment (tomtau) [inline app/export.go:51]:
```suggestion
	if err != nil {
		return servertypes.ExportedApp{}, err
	}
	return servertypes.ExportedApp{
		AppState:        appState,
		Validators:      validators,
		Height:          height,
		ConsensusParams: app.BaseApp.GetConsensusParams(ctx),
	}, nil
```

File: app/export.go [hunk 3]
```diff
@@ -40,12 +40,15 @@ func (app *ChainApp) ExportAppStateAndValidators(
 	}
 
 	validators, err := staking.WriteValidators(ctx, app.StakingKeeper)
+	if err != nil { 
+		return servertypes.ExportedApp{}, err
+	}
```
Comment (tomtau) [inline app/export.go:45]:
```suggestion
	if err != nil {
		return servertypes.ExportedApp{}, err
	}
```

**Review by tomtau — APPROVED** [review]

**Comment by tomtau** [comment]
will update to a newer Cosmos SDK in a subsequent PR

**Pull request merged**